File: core-banking-prototype-laravel-main/core-banking-prototype-laravel-main/sdks/python/finaegis/resources/exchange_rates.py
"""
Exchange rates resource for the FinAegis SDK
"""

from typing import Dict, Any
from ..types import ExchangeRate, PaginatedResponse
from .base import BaseResource


class ExchangeRatesResource(BaseResource):
    """Manage exchange rates in the FinAegis platform."""
    
    def list(self, page: int = 1, per_page: int = 20) -> PaginatedResponse:
        """
        List all exchange rates.
        
        Args:
            page: Page number
            per_page: Items per page
            
        Returns:
            PaginatedResponse containing ExchangeRate objects
        """
        response = self._get('/exchange-rates', params={'page': page, 'per_page': per_page})
        return PaginatedResponse.from_dict(response, ExchangeRate)
    
    def get(self, from_asset: str, to_asset: str) -> ExchangeRate:
        """
        Get exchange rate between two assets.
        
        Args:
            from_asset: Source asset code
            to_asset: Target asset code
            
        Returns:
            ExchangeRate object
        """
        response = self._get(f'/exchange-rates/{from_asset}/{to_asset}')
        return ExchangeRate.from_dict(response['data'])
    
    def convert(self, from_asset: str, to_asset: str, amount: float) -> Dict[str, Any]:
        """
        Convert amount between two assets.
        
        Args:
            from_asset: Source asset code
            to_asset: Target asset code
            amount: Amount to convert
            
        Returns:
            Dictionary with conversion details
        """
        response = self._get(
            f'/exchange-rates/{from_asset}/{to_asset}/convert',
            params={'amount': amount}
        )
        return response['data']
    
    def refresh(self) -> Dict[str, Any]:
        """
        Refresh all exchange rates.
        
        Returns:
            Dictionary with refresh status
        """
        response = self._post('/exchange-rates/refresh')
        return response['data']

File: core-banking-prototype-laravel-main/core-banking-prototype-laravel-main/sdks/php/src/Resources/ExchangeRates.php
<?php

namespace FinAegis\Resources;

use FinAegis\Models\ExchangeRate;
use FinAegis\Models\PaginatedResponse;

class ExchangeRates extends BaseResource
{
    /**
     * List all exchange rates.
     *
     * @param  int  $page  Page number
     * @param  int  $perPage  Items per page
     */
    public function list(int $page = 1, int $perPage = 20): PaginatedResponse
    {
        $response = $this->get('/exchange-rates', ['page' => $page, 'per_page' => $perPage]);

        return new PaginatedResponse($response, ExchangeRate::class);
    }

    /**
     * Get exchange rate between two assets.
     *
     * @param  string  $fromAsset  Source asset code
     * @param  string  $toAsset  Target asset code
     */
    public function get(string $fromAsset, string $toAsset): ExchangeRate
    {
        $response = $this->get("/exchange-rates/{$fromAsset}/{$toAsset}");

        return new ExchangeRate($response['data']);
    }

    /**
     * Convert amount between two assets.
     *
     * @param  string  $fromAsset  Source asset code
     * @param  string  $toAsset  Target asset code
     * @param  float  $amount  Amount to convert
     */
    public function convert(string $fromAsset, string $toAsset, float $amount): array
    {
        $response = $this->get("/exchange-rates/{$fromAsset}/{$toAsset}/convert", ['amount' => $amount]);

        return $response['data'];
    }

    /**
     * Refresh all exchange rates.
     */
    public function refresh(): array
    {
        $response = $this->post('/exchange-rates/refresh');

        return $response['data'];
    }
}
